src/lib.rs

```diff
@@ -31,17 +43,36 @@ fn relaunch() -> Result<Child, Error> {
 }
 
 fn pid1_handling(child: Option<Child>) -> ! {
-    let child = child.map(|x| x.id());
     let mut signals = Signals::new([SIGTERM, SIGINT, SIGCHLD]).unwrap();
+
+    let child = child.map(|x| x.id());
+    struct ExitStatus {
+        pid: Pid,
+        exit_code: i32,
+    }
+
     loop {
-        for signal in signals.pending() {
+        for signal in signals.forever() {
```

Comment:
This fixes the busy loop issue. Earlier the above loop cause 100% cpu consumption.

Reply:
Doh! I didn't read the API docs well enough it seems.